bundles/AdminBundle/Controller/Admin/Document/SnippetController.php

```diff
@@ -106,7 +108,31 @@ public function saveAction(Request $request)
         try {
             if ($request->get('id')) {
                 $snippet = Document\Snippet::getById($request->get('id'));
-                $snippet = $this->getLatestVersion($snippet);
+
+                // check if there's a document in session which should be used as data-source
+                // see also PageController::clearEditableDataAction() | this is necessary to reset all fields and to get rid of
+                // outdated and unused data elements in this document (eg. entries of area-blocks)
+                $snippetSession = Session::useSession(function (AttributeBagInterface $session) use ($snippet) {
```

Comment:
@dvesh3 To avoid duplicate code we should move this function to `DocumentControllerBase` and use it in both `PageController` and `SnippetController`, that should be easily possible as far as I can see. 
Thanks!

Reply:
done!